Mathlib/Algebra/Module/Submodule/Bilinear.lean

```diff
@@ -56,13 +56,15 @@ theorem map₂_span_span (f : M →ₗ[R] N →ₗ[R] P) (s : Set M) (t : Set N)
   apply le_antisymm
   · rw [map₂_le]
     apply @span_induction' R M _ _ _ s
-    intro a ha
-    apply @span_induction' R N _ _ _ t
-    intro b hb
-    exact subset_span ⟨_, ‹_›, _, ‹_›, rfl⟩
-    all_goals intros; simp only [*, add_mem, smul_mem, zero_mem, _root_.map_zero, map_add,
-                                 LinearMap.zero_apply, LinearMap.add_apply, LinearMap.smul_apply,
-                                 map_smul]
+    on_goal 1 =>
+      intro a ha
+      apply @span_induction' R N _ _ _ t
+      on_goal 1 => intro b hb
+      · exact subset_span ⟨_, ‹_›, _, ‹_›, rfl⟩
```

Comment:
I'm surprised by the nested `on_goal` here, and can't readily guess what is going on here.

Reply:
I removed the nesting: the first `on_goal` acted on the first goal, while "storing" the remaining ones.  With the release of `on_goal`, the following `all_goals` took over the rest of the proof.  The nested `on_goal` could be (and has been) replaced by a focusing dot.